zeebe/exporters/camunda-exporter/src/test/java/io/camunda/exporter/archiver/ElasticsearchRepositoryIT.java

```diff
@@ -0,0 +1,255 @@
+        .as("only the third document is remaining")
+        .hasSize(1)
+        .first()
+        .extracting(Hit::id)
+        .isEqualTo("3");
+  }
+
+  @Test
+  void shouldSetIndexLifeCycle() throws IOException {
+    // given
+    final var indexName = UUID.randomUUID().toString();
+    final var repository = createRepository();
+    testClient.indices().create(r -> r.index(indexName));
+    final var initialLifecycle =
+        testClient
+            .indices()
+            .getSettings(r -> r.index(indexName))
+            .get(indexName)
+            .settings()
+            .lifecycle();
+    assertThat(initialLifecycle).isNull();
+    retention.setEnabled(true);
+    retention.setPolicyName("operate_delete_archived_indices");
+    putLifecyclePolicy();
+
+    // when
+    final var result = repository.setIndexLifeCycle(indexName);
+
+    // then
+    assertThat(result).succeedsWithin(Duration.ofSeconds(30));
+    final var actualLifecycle =
+        testClient
+            .indices()
+            .getSettings(r -> r.index(indexName))
+            .get(indexName)
+            .settings()
+            .index()
+            .lifecycle();
+    assertThat(actualLifecycle)
+        .isNotNull()
+        .extracting(IndexSettingsLifecycle::name)
+        .isEqualTo("operate_delete_archived_indices");
+  }
+
+  @Test
+  void shouldNotSetIndexLifecycleIfRetentionIsDisabled() throws IOException {
+    // given
+    final var indexName = UUID.randomUUID().toString();
+    final var repository = createRepository();
+    testClient.indices().create(r -> r.index(indexName));
+    final var initialLifecycle =
+        testClient
+            .indices()
+            .getSettings(r -> r.index(indexName))
+            .get(indexName)
+            .settings()
+            .lifecycle();
+    assertThat(initialLifecycle).isNull();
+    retention.setEnabled(false);
+    retention.setPolicyName("operate_delete_archived_indices");
+    putLifecyclePolicy();
+
+    // when
+    final var result = repository.setIndexLifeCycle(indexName);
+
+    // then
+    assertThat(result).succeedsWithin(Duration.ZERO);
+    final var actualLifecycle =
+        testClient
+            .indices()
+            .getSettings(r -> r.index(indexName))
+            .get(indexName)
+            .settings()
+            .index()
+            .lifecycle();
+    assertThat(actualLifecycle).isNull();
+  }
+
+  @Test
+  void shouldReindexDocuments() throws IOException {
+    // given
+    final var sourceIndexName = UUID.randomUUID().toString();
+    final var destIndexName = UUID.randomUUID().toString();
+    final var repository = createRepository();
+    final var documents =
+        List.of(
+            new TestDocument("1", "2024-01-01"),
+            new TestDocument("2", "2024-01-04"),
+            new TestDocument("3", "2024-01-02"));
+    documents.forEach(doc -> index(sourceIndexName, doc));
+    testClient.indices().refresh(r -> r.index(sourceIndexName));
```

Comment:
❓ I am missing some knowledge of ES, why do we need to refresh here?

Reply:
When you index documents in ES, the request returns when the server has accepted it (so persisted somewhere). However, because there's some caching happening in ES (I'm also not an expert, so I can't go into details confidently :sweat_smile:), it may take a bit for writes to become visible to clients. Refresh forces that to happen in a synchronous way, so we can guarantee the documents are visible afterwards.

Without it, the test would become flaky, basically.

Reply:
oh I see, thanks for the explanation 🙌